parlai/agents/seq2seq/modules.py

```diff
@@ -66,8 +69,51 @@ def __init__(self, opt, num_features,
                 padding_idx=self.NULL_IDX,
                 attn_type=opt['attention'])
 
+        self.beam_dump = opt['beam_dump']
+        if self.beam_dump > 0.0:
+            self.dict = DictionaryAgent(opt)
+            self.beam_dump_filecnt = 0
+            self.beam_dump_path = os.path.join(os.path.dirname(opt['model_file']), 'beam_dump')
+            if not os.path.exists(self.beam_dump_path):
+                os.makedirs(self.beam_dump_path)
+
+        if opt['no_cuda'] is True:
+            self.beam_device = 'cpu'
+        else:
+            self.beam_device = 'cuda'
```

Comment:
Please infer this automatically in the beam code (if user send cuda tensor, move scores to cuda)